### DISCUSSION TITLE
Docker automation

### DISCUSSION THREAD

**Discussion by kmax9981** [ptr727/PlexCleaner#392] [Q&A] [answered]
I have been running plexcleaner on a windows machine using a powershell script and scheduled tasks to run the script on a schedule.  I am moving my automation tools to an ubuntu server.  I used the interactive shell to run plexcleaner with my json file as a test and it ran fine.  Is there a good way to script this to run on a schedule.  The issue I see is the interactive shell with docker and not being there to interact and exit the shell when finished.

**Comment by ptr727** [comment, marked as answer]
No need to run interactive in docker, instead of the `/bin/bash` from the example in readme just run `/PlexCleaner/...` as the command, it is as if you are running the executable but it is hosted in docker.

**Reply by kmax9981** [reply]
Thanks, I just figured this out and came back to comment lol, I appreciate you!

**Reply by kmax9981** [reply]
so something I am seeing now is if I run docker with the -d to detach, it runs fine, however it doesn't write to the log file.  I have a volume mounted to the docker config (--volume /opt/plexcleaner/config:/config:rw) this is where I point the log file to, however it doesn't create it. If I run it in interactive mode it does create and write to the log file.

**Reply by ptr727** [reply]
What does docker logs say? Are the file permissions different? Do you run as root or a non-root user? I have not tested this scenario myself, but I'd expect it to be no different, but never know without some debugging.

**Reply by kmax9981** [reply]
I am running it with sudo, so root user.  The output of docker logs plexcleaner is below

14:10:11 [INF] <1> Loading settings from : "/config/PlexCleaner.json"
14:10:11 [INF] <1> Logging output to : "/config/plexcleaner.log"
14:10:11 [INF] <1> Application Version : "PlexCleaner : 3.9.28.59862 (Release)"
14:10:11 [INF] <1> Runtime Version : ".NET 8.0.5" : "ubuntu.24.04-x64"
14:10:11 [INF] <1> OS Version : "Ubuntu 24.04 LTS"
14:10:11 [INF] <1> Build Date : 06/21/2024 13:52:28
14:10:11 [INF] <1> Parallel Processing: True : Thread Count: 4, Processor Count: 8

so it shows that it is logging output to /config/plexcleaner.log which is a mounted folder at /opt/plexcleaner/config

If I goto /opt/plexcleaner/config there is no log, however if I create the log with touch then it start writing to the file

However if I create the file before running the docker command it deletes the file and then will not create the new file until I create it again using touch

**Reply by kmax9981** [reply]
I removed --logwarning from the docker command and it began logging properly

**Reply by ptr727** [reply]
Using `--logappend` will add to existing file.  
Logger is created [here](https://github.com/ptr727/PlexCleaner/blob/main/PlexCleaner/Program.cs#L235)

If you say you run using `sudo`, that does not really relate to how the container is executing, maybe `sudo docker` because you did not add your account to the `docker` group to not need root for managing docker?

**Reply by ptr727** [reply]
Maybe it is because there were no warnings being logged, so no output generated until a warning is emitted?

**Comment by kmax9981** [comment]
Sorry, the uid is my user, so I guess it is running with my permissions

**Comment by kmax9981** [comment]
Ok, thanks for the clarification.
